app/js/components/mixins/ModalEffects.react.js

```diff
@@ -0,0 +1,46 @@
+import React from 'react'
+
+let ModalEffects = InnerComponent => class extends React.Component {
```

Comment:
@andela-ashuaib They are called Higher Order functions. It's not 'bad practice' since they're relatively new and were introduced only in es6. They replace traditional use of mixins (support for the latter would soon be fully dropped from react API).
You should look it up 👍